packages/composites/augur/src/dataProviders/sportsdataio.ts

```diff
@@ -5,40 +5,137 @@ import { BigNumber, ethers } from 'ethers'
 import { CreateEvent } from '../methods/createMarkets'
 import { ResolveEvent } from '../methods/resolveMarkets'
 
+export const SPORTS_SUPPORTED = ['nfl', 'ncaa-fb']
+
 interface NFLEvent {
-  PointSpread: number
+  PointSpread: number | null
   Date: string | null
+  Day: string
   GlobalGameID: number
   GlobalAwayTeamID: number
   GlobalHomeTeamID: number
   Status: string
 }
 
-type SportsdataioNFLSchedule = NFLEvent[]
+interface CommonSchedule {
+  Date: string
+  GameID: number
+  AwayTeamID: number
+  HomeTeamID: number
+  Status: string
+  PointSpread: number | null
+}
+
+interface NFLScores {
+  Date: string | null
+  Day: string
+  GlobalGameID: number
+  GlobalAwayTeamID: number
+  GlobalHomeTeamID: number
+  Status: string
+  AwayScore: number | null
+  HomeScore: number | null
+}
+
+interface CFBGames {
+  Day: string
+  DateTime: string | null
+  GlobalGameID: number
+  GlobalAwayTeamID: number
+  GlobalHomeTeamID: number
+  Status: string
+  AwayTeamScore: number | null
+  HomeTeamScore: number | null
+  PointSpread: number | null
+}
+
+interface CommonScores {
+  Date: string
+  GameID: number
+  AwayTeamID: number
+  HomeTeamID: number
+  Status: string
+  AwayScore: number | null
+  HomeScore: number | null
+}
 
-const getNFLSchedule = async (id: string, season: string, sportsdataExec: Execute): Promise<SportsdataioNFLSchedule> => {
+const getSchedule = async (id: string, sport: string, season: string, exec: Execute): Promise<CommonSchedule[]> => {
   const input = {
     id,
     data: {
-      sport: 'nfl',
+      sport,
       endpoint: 'schedule',
       season
     }
   }
-  const response = await sportsdataExec(input)
+  const response = await exec(input)
+  const filtered = (response.result as { GlobalGameID: number }[])
+    .filter(event => event.GlobalGameID != 0)
+
+  switch (sport) {
+    case 'nfl': {
+      return (filtered as NFLEvent[]).map(event => ({
+        Date: event.Date || event.Day,
+        GameID: event.GlobalGameID,
+        AwayTeamID: event.GlobalAwayTeamID,
+        HomeTeamID: event.GlobalHomeTeamID,
+        Status: event.Status,
+        PointSpread: event.PointSpread,
+      }))
+    }
+    case 'ncaa-fb': {
+      return (filtered as CFBGames[]).map(event => ({
+        Date: event.DateTime || event.Day,
+        GameID: event.GlobalGameID,
+        AwayTeamID: event.GlobalAwayTeamID,
+        HomeTeamID: event.GlobalHomeTeamID,
+        Status: event.Status,
+        PointSpread: event.PointSpread
+      }))
+    }
```

Comment:
Should we have a default to error?

Reply:
We could throw an error here, but the sport should already be validated to be in `SPORTS_SUPPORTED` on L156. I can add in just a default handling just in case.